### ISSUE TITLE
Next improvements

### ISSUE THREAD

**Issue by matekrk** [matekrk/Delayed_generalization#23]
1. Update todo what is done (remove) and what is not
2. Proceed with next todo steps
3. enhance visualization section - some visualizations plots are being written in individual training scripts (for instance in phenomena/grokking/training/train_modular.py def plot_training_curves but there are many more examples!)
4. Make sure that all training scripts follow up more less the same pattern (use utils, use optimization toolbox, wandb support, visualizations etc.)
5. There is a bit chaos with files - (a) make sure that all data relevant things are in data/ subdir, all models relevant are in models/ and all optimization toolbox are in optimization/ -> also there are some files like phenomena/simplicity_bias/celeba/data/generate_synthetic_celeba.py that I believe are not being used because the data script is in other place <but there are more examples like that>
6. Optimization dir is still not well written - there are some subdirs such as batch_strategies (take data in specific order/negative sampling/discard some data with replacement based on some metrics, progressive batch size, momentum and adaptive lookaheads tools, advanced regularization (vanilla L_p regularization but also internal tricks), schedulers and warmup.
7. use optimizer_utils and if relevant image_analysis from utils to improve the granulality of analysis
8. in getting started make sure there is curated list of what is already achieved (and in implementation_status.md) with relevant commands running training script on: colored mnist, cifar10c, cifar100c, continual ciifar100, grokked modular arithmetic data, waterbirds simplicity bias, celeba simplicity bias, nlp. vanilla training but also with wandb support with optimization toolbox and detailed analysis if relevant (such as predominant color)
9. celeba: additional_attrs is good idea, but also include option "all" which basically takes all other attributes without listing them all. also these attributes that are being included (or 'all') should also be in output_dir so that user knows what exactly is in that subset
10. use data attribution (already some code in data_attribution/) as other form of analysis (trak/gradcam) to the training methods

go over all .md files and update (but do not use any "new feature" - just pure information what is supported)
